### DISCUSSION TITLE
Api testing

### DISCUSSION THREAD

**Discussion by AnuSam123** [pactumjs/pactum#152] [General]
Can this tool have libs and support for Gitlab CI/CD pipeline

**Comment by ASaiAnudeep** [comment]
**PactumJS** is not tightly integrated with any CI/CD platform. We can write tests with any chosen test framework and run it any CI/CD environment.

**Reply by AnuSam123** [reply]
Thank you!

**Reply by AnuSam123** [reply]
I have couple of  more question too, thanks in advance!

Can this tool have libraries to provide test results (json) to JPEG screenshots/any other screenshots as an evidence?
Can I re use the existing JS automation api test cases which are already developed (no framework is existing, just returned in JS language)in this framework or do I need to write it from scratch?

**Reply by ASaiAnudeep** [reply]
**Can this tool have libraries to provide test results (json) to JPEG screenshots/any other screenshots as an evidence?**
Yes, it can generate test results in json or html formats. Ideally the generation of test results boils down to the underlying test-framework used to run it like [mocha](https://mochajs.org/#reporters). Instead of screenshots you can integrate it with reporters like [mochawesome](https://www.npmjs.com/package/mochawesome).

**Can I re use the existing JS automation api test cases which are already developed (no framework is existing, just returned in JS language)in this framework or do I need to write it from scratch?**
I think you can slowly migrate tests to this library where you can have both the tests in the same repo or so. Instead of writing it plain js, better integrate it with **mocha** or **cucumber**.

**Reply by AnuSam123** [reply]
That's great

**Reply by AnuSam123** [reply]
Do you think we can do UI testing as well using pactumjs, or do we need to club with other frameworks like Cypress or any other tools to be able to do UI testing?

**Reply by ASaiAnudeep** [reply]
You need to club with other frameworks for UI testing.

**Reply by AnuSam123** [reply]
Sorry, you mean to say pactumjs will work for UI automation and for API testing we need to club with other frameworks or vice versa.
i was thinking pactumjs is good framework for rest api automation

**Reply by ASaiAnudeep** [reply]
It was a typo. I just updated my comment. PactumJS is designed for API Automation Testing.

**Reply by AnuSam123** [reply]
Thank you so much Anudeep!

**Comment by diegozklein** [comment]
So, I can integrate Pectum tests with Azure DevOps with no problems?

**Reply by ASaiAnudeep** [reply]
Yes, I have been running pactum in Azure DevOps pipelines without any issues for a long time.

**Reply by diegozklein** [reply]
Oh Nice, My company is starting to automate api tests and we're probably going to use pactum. Thanks for the fast answer btw.
